src/plugins/identity/identity.js

```diff
@@ -31,16 +32,30 @@ export type IdentitySpec = {|
 |};
 
 /**
- * Create a new node representing an identity.
+ * Internal method for validating a username.
+ *
+ * Returns the username with any leading @ symbol stripped.
+ * Throws an error if the username is invalid.
  */
-export function identityNode(identity: Identity): Node {
+function validateUsername(username: string): Username {
   const re = new RegExp(USERNAME_PATTERN);
-  const match = re.exec(identity.username);
+  const match = re.exec(username);
   if (match == null) {
-    throw new Error(`Invalid username: ${identity.username}`);
+    throw new Error(`Invalid username: ${username}`);
   }
-  const username = match[1];
-  const address = NodeAddress.append(nodePrefix, username);
+  return match[1];
+}
+
+/**
+ * Create a new node representing an identity.
+ */
+export function identityNode(identity: Identity): Node {
+  const username = validateUsername(identity.username);
+  const address = identityAddress(username);
   const description = `@${username}`;
   return {address, timestampMs: null, description};
 }
+
+export function identityAddress(username: Username): NodeAddressT {
```

Comment:
nit: since its a function, would be more readable as "getIdentityAddress" or "getAddressForUsername". "identityAddress" sounds more like a name for a variable.

Reply:
See above.